src/cryptonote_basic/account.cpp

```diff
@@ -50,6 +51,24 @@ DISABLE_VS_WARNINGS(4244 4345)
 
   namespace cryptonote
 {
+ account_keys& account_keys::operator=(const account_keys &keys) {
+      m_account_address  = keys.m_account_address;
+      m_spend_secret_key = keys.m_spend_secret_key;
+      m_view_secret_key  = keys.m_view_secret_key;
+      m_multisig_keys    = keys.m_multisig_keys;
+      m_device           = keys.m_device;
+      return *this;
+    }
+  //-----------------------------------------------------------------
+  hw::Device& account_keys::get_device() const  {
+    return *m_device;
+  }
+  //-----------------------------------------------------------------
+  void account_keys::set_device( hw::Device &device)  {
+    m_device = &device;
```

Comment:
Might be too late to make this change, but it seems like it make more sense to have all of the account keys owned by the device. The different devices could then have different strategies - the default would "own" the keys, the hardware ledger would "own" the spend but not the view key, etc. With some mixed modes too. Probably a big change at this point though. In this design the device would _not_ be a singleton.

Reply:
Yes Its a little bit too late :)

The today approach is to leave the key to the global wallet and not to the device. But your suggestion need to be investigated. I note it for later.